src/Rules/Generics/GenericAncestorsCheck.php

```diff
@@ -98,7 +98,7 @@ public function check(
 				$messages[] = RuleErrorBuilder::message(sprintf($invalidTypeMessage, $referencedClass))->build();
 			}
 
-			$variance = TemplateTypeVariance::createInvariant();
+			$variance = TemplateTypeVariance::createStatic();
```

Comment:
What about this? What does it change? What's the impact? Shouldn't it be done under the feature toggle?

Reply:
When I fixed the invariance composition, this check reported _all_ template types in `@extends` as invariant. Therefore, this would not be allowed:

```php
/** @template-covariant T */
interface Foo {}

/**
 * @template-covariant T
 * @implements Foo<T>
 */
class Bar implements Foo {}
```

I can add the feature toggle but I don't think it's necessary. This change has no practical impact because "static" variance has had the same composition logic as invariant until this PR. It even feels semantically more correct to me: the initial position in `@extends` is not invariance, it is _nothing_, similar to other static contexts like static methods or the constructor.

Reply:
Alright, great 😊